packages/ui/src/components/va-button/hooks/useButtonTextColor.ts

```diff
@@ -1,21 +1,45 @@
 import { computed, Ref, ComputedRef, getCurrentInstance } from 'vue'
 
 import { useColors } from '../../../composables'
+import { isServer } from '../../../utils/ssr'
 
 import { ButtonPropsTypes } from '../types'
 
-type UseButtonTextColor = (
-  textColorComputed: Ref<string>,
-  colorComputed: Ref<string>,
-  isPressed: Ref<boolean>,
-  isHovered: Ref<boolean>,
-) => ComputedRef<{
+type ButtonTextColorStyles = {
   color: string
   background?: string
   opacity?: number
   'background-clip'?: 'text',
   '-webkit-background-clip'?: 'text',
-}>
+}
+
+type UseButtonTextColor = (
+  textColorComputed: Ref<string>,
+  colorComputed: Ref<string>,
+  isPressed: Ref<boolean>,
+  isHovered: Ref<boolean>,
+) => ComputedRef<ButtonTextColorStyles>
+
+/*
+* Inverts opacity depending on browser (webkit/chromium)
+* TODO: Check if Safari <16 version will be less then 1% and remove function
+*
+* @param {number} opacity - Opacity value
+* @returns {number} Inverted opacity value
+* */
+const getOpacity = (opacity: number): number => {
+  if (isServer()) { return opacity }
+
+  if (/^((?!chrome|android).)*safari/i.test(window?.navigator?.userAgent)) {
+    const isLatestSafari = /(version.)15|16/i.test(window?.navigator?.userAgent)
+
+    if (!isLatestSafari) {
+      return opacity < 1 ? 1 - opacity : opacity
```

Comment:
wtf?

Looks like it is impossible to have opacity 0 here. Is that ok?

Should be more documented, cuz for now it is magic.

Reply:
Oopsie, forgot about it. Will fix it. :)